internal/service/searchindex/data_source_search_indexes.go

```diff
@@ -35,7 +35,7 @@ func PluralDataSource() *schema.Resource {
 				Type:     schema.TypeList,
 				Computed: true,
 				Elem: &schema.Resource{
-					Schema: returnSearchIndexSchema(),
+					Schema: returnSearchIndexDSSchema(),
```

Comment:
plural data source  using the resource schema instead of the singular data source

@marcosuma you're right about adding comments to help reviewers, added a couple of them